### PR TITLE
Mat 4169 umls logout

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-4169](https://jira.cms.gov/browse/MAT-4169)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/api/useTerminologyServiceApi.ts [hunk 1]
```diff
@@ -48,6 +48,27 @@ export class TerminologyServiceApi {
...
+    const resp = await axios
+      .delete(`${baseUrl}/vsac/umls-credentials`, {
+        headers: {
+          Authorization: `Bearer ${this.getAccessToken()}`,
+          "Content-Type": "text/plain",
+        },
+        timeout: 15000,
+      })
+      .then((resp) => {
+        if (resp.status === 200) {
```
Comment (sb-prateekkeerthi) [inline src/api/useTerminologyServiceApi.ts]: async logoutUMLS(): Promise<boolean> {
  try {
    const baseUrl = await getServiceUrl();
    const resp = await axios.delete(`${baseUrl}/vsac/umls-credentials`, {
      headers: {
        Authorization: Bearer ${this.getAccessToken()},
        "Content-Type": "text/plain",
      },
      timeout: 15000,
    });

    // Check response status and return true if successful
    return resp.status === 200;
  } catch (error) {
    // Log the error or handle it as needed
    console.error("Logout failed:", error);
    return false;
  }
}

--> I think you can use the above optimized code which removes the unnecessary .then .catch
--> this also ensures proper error handling and removes redundant return statements

**Review by sb-prateekkeerthi — CHANGES_REQUESTED** [review]

File: src/api/useTerminologyServiceApi.ts (hunk 1 above)
Comment (sb-cecilialiu) [reply src/api/useTerminologyServiceApi.ts]: @sb-prateekkeerthi I added error log, but i think .then and .catch are ok, same thing as try...catch, plus, it's more consistent with other service calls.

**Review by sb-prateekkeerthi — APPROVED** [review]
